datagram-socket/src/socket_stats.rs

```diff
@@ -242,6 +250,18 @@ impl QuicAuditStats {
     pub fn max_loss_pct(&self) -> u8 {
         self.max_loss_pct.load(Ordering::Acquire)
     }
+
+    #[inline]
+    #[cfg(target_os = "linux")]
+    pub fn set_initial_so_mark_data(&self, value: Option<Vec<u8>>) {
+        *self.initial_so_mark.write().unwrap() = value;
```

Comment:
Since this is supposed to be the initial, should this check that the value was not already set before allowing the write?

Reply:
Good catch 😅